### DISCUSSION TITLE
M5StickCPlus Display Messed Up (Custom Firmware)

### DISCUSSION THREAD

**Discussion by efaden** [arendst/Tasmota#15096] [Q&A] [answered]
I have a M5Stick C Plus running a firmware I built with BLE and LVGL.  I did this by making a custom partition CSV and then using the following build

`
[env:tasmota32-cplus]
extends                 = env:tasmota32-lvgl
board                   = esp32-m5stickcplus
build_flags             = ${env:tasmota32-lvgl.build_flags} -DFIRMWARE_BLUETOOTH
lib_extra_dirs          = lib/libesp32, lib/libesp32_lvgl, lib/libesp32_div, lib/lib_basic, lib/lib_i2c, lib/lib_ssl, lib/lib_display
`

which references my custom board which is basically just the 4M ESP32 board referencing my custom partition map.

I also am using a custom user config override of

`
#define USE_LVGL
#define USE_DISPLAY_LVGL_ONLY
#define USE_DISPLAY
#define USE_XPT2046
#define USE_UNIVERSAL_DISPLAY
#undef USE_DISPLAY_MODES1TO5
#undef USE_DISPLAY_LCD
#undef USE_DISPLAY_SSD1306
#undef USE_DISPLAY_MATRIX
#undef USE_DISPLAY_SEVENSEG
`

I am using the AXP.be as preinit.be and the display.ini from the M5Stick.

The display is however.... not working correctly.  On boot up I get what appears to be a blurry display with the word tasmota partially written in the corner backwards.

![PXL_20220310_124341160](https://user-images.githubusercontent.com/12599300/157674282-756d2f08-6283-45c0-b1e7-0d0bed3ed473.jpg)
![PXL_20220310_124535167](https://user-images.githubusercontent.com/12599300/157674285-131233ab-e97b-4091-81db-6b33d159a57f.jpg)

**Comment by efaden** [comment]
Turns out it isn't a correct display.ini.  Not sure exactly what to fill into the display.ini. but it is some sort of ST7789v2 instead of the ST7735S display ini from the older stick.

**Comment by Jason2866** [comment, marked as answer]
Have you tried https://github.com/arendst/Tasmota/blob/development/tasmota/displaydesc/ST7789_display.ini

**Reply by efaden** [reply]
It's not the correct dimensions.  But yes.... at least the display no longer seems to have lines or whatever.  Just solid black.  Currently I am using that one with the dimensions changed

**Reply by efaden** [reply]
Getting really close..
![PXL_20220310_154136292](https://user-images.githubusercontent.com/12599300/157698226-6d7d79dd-d664-40aa-8024-00ee51ea36e4.jpeg)
.

**Reply by tfmt100** [reply]
Hi! Did you solve that blurry stripe in yours?
